### PR TITLE
Now possible to use file path as `privateKey` value

### PR DESCRIPTION
**Pull request by prometheas**
Found that this is a useful feature. Sorry it's a wee bit kludgy (performing the require in a nested code block and whatnot), but I'm in a rush at work and simply wanted to illustrate the idea via PR.

### CODE REVIEW

**Comment by mscdex** [comment]
I think this is a bit too magical for my taste.

**Comment by prometheas** [comment]
For what it's worth, there's absolutely NO chance that a private key value could ever match a file path… but taste is taste, so I'll not over advocate.

Still. It does seem like a sensible use case, if I may say so… perhaps activated by an explicit config flag?

**Comment by unknownbrackets** [comment]
What about `privateKeyFile` just for convenience?  Can make gulp configs, etc. simpler still.

Also, would it be too magical if instead of "Unsupported key format" it checked if a file exists and said "did you mean privateKeyFile"?

-[Unknown]

File: lib/client.js [hunk 1]
```diff
@@ -142,6 +142,12 @@ Client.prototype.connect = function(cfg) {
   this._remoteVer = undefined;
 
   if (this.config.privateKey) {
+
+    var fs = require('fs');
+    if (fs.existsSync(this.config.privateKey)) {
```
Comment (steelbrain) [inline lib/client.js]: how about a `require('path').isAbsolute(this.config.privateKey)` check here?

**Comment by mscdex** [comment]
Thanks for the PR, but I still think at this point I'd rather choose to keep it simple and just support the one interface. IMHO this kind of abstraction is best handled outside of this module.

**Pull request closed without merging**